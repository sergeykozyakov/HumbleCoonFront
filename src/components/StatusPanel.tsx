import React from 'react';

import {
  Typography,
  StyleRules,
  Theme,
  Grid,
} from '@material-ui/core';

import { withStyles } from '@material-ui/core/styles';

import WarningIcon from '@material-ui/icons/Warning';
import CheckIcon from '@material-ui/icons/Check';

interface IProps {
  message: string;
  isError?: boolean;
  classes: Record<string, string>;
}

const styles = (theme: Theme): StyleRules<string, any> => ({
  text: {
    paddingTop: theme.spacing(1),
    paddingLeft: theme.spacing(1),
  },
});

const StatusPanel = (props: IProps): JSX.Element => {
  const { message, isError = false, classes = {} } = props;

  return (
    <Grid container>
      <Grid item>
        {isError
          ? <WarningIcon color="secondary" fontSize="large" />
          : <CheckIcon fontSize="large" />}
      </Grid>
      <Grid item>
        <Typography
          color={isError ? 'secondary' : undefined}
          className={classes.text}
        >
          {message}
        </Typography>
      </Grid>
    </Grid>
  );
};

export default withStyles(styles)(StatusPanel);
