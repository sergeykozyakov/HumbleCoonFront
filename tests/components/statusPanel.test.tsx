import React from 'react';

import { mount, ReactWrapper } from 'enzyme';

import '../init';

import StatusPanel from '../../src/components/StatusPanel';

interface IProps {
  message: any;
  isError?: boolean;
};

describe('Components / StatusPanel', () => {
  it('renders correctly', () => {
    const StatusPanelComponent: ReactWrapper = mount(<StatusPanel message="" />);
    expect(StatusPanelComponent).toMatchSnapshot();
  });

  it('renders component correctly with null message', () => {
    const props: IProps = {
      message: null,
    };

    const StatusPanelComponent: ReactWrapper = mount(<StatusPanel {...props} />);
    expect(StatusPanelComponent.prop('message')).toEqual(null);
  });

  it('renders component correctly without error flag', () => {
    const props: IProps = {
      message: 'test',
    };

    const StatusPanelComponent: ReactWrapper = mount(<StatusPanel {...props} />);

    expect(StatusPanelComponent.prop('message')).toEqual('test');
    expect(StatusPanelComponent.prop('isError')).toEqual(undefined);
  });

  it('renders component correctly with error flag', () => {
    const props: IProps = {
      message: 'test',
      isError: true,
    };

    const StatusPanelComponent: ReactWrapper = mount(<StatusPanel {...props} />);

    expect(StatusPanelComponent.prop('message')).toEqual('test');
    expect(StatusPanelComponent.prop('isError')).toEqual(true);
  });
});
