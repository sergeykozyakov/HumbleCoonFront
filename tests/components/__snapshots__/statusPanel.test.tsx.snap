// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components / StatusPanel renders correctly 1`] = `ReactWrapper {}`;
